apps/synapse-portal-framework/src/components/DetailsPage/DetailsPageContentLayout.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import { scrollToWithOffset } from '../../utils'
+import {
+  DetailsPageSectionLayout,
+  DetailsPageSectionLayoutType,
+} from './DetailsPageSectionLayout'
+
+export type DetailsPageContentType = DetailsPageSectionLayoutType[]
+
+export const handleMenuClick = (id: string) => {
+  window.history.pushState(null, '', `#${id}`)
+  // An ID may have spaces, so use [id="..."] style selector
+  // #... will treat spaces as a query for descendants
+  const query = `[id="${id}"]`
+  const wrapper = document.querySelector<HTMLElement>(query)
+  if (wrapper) {
+    scrollToWithOffset(wrapper)
+  } else {
+    console.error(`Could not find id ${id} in document (query: ${query})`)
+  }
+}
+
+export function DetailsPageContent(props: {
```

Comment:
Given an array of content (type copied below for convenience), this component will render the DetailsPage sticky menu and a section layout component that includes the component to render, and may also include a title and hash link.

```ts
/**
 * Describes a section of a portals DetailsPage. Using this interface allows us to render
 * both the side menu and the content based on a single definition.
 */
export type DetailsPageSectionLayoutType = {
  /** The title to use for the button and the section. If omitted, the section will not appear in the side menu. */
  title?: string
  /** If true, hide the title from the section content. The button will still appear in the menu */
  hideTitle?: boolean
  /** The unique, persistent ID to use for linking to the section */
  id: string
  /** The contents of the section */
  element: React.ReactNode
  helpText?: string
}
```